packages/apptive_grid_form/lib/src/apptive_grid_form.dart

```diff
@@ -795,3 +590,389 @@ class ApptiveGridFormDataState extends State<ApptiveGridFormData> {
+  /// Creates a new error widget
+  const FormErrorWidget({
+    super.key,
+    required this.error,
+    required this.padding,
+    required this.didTapBackButton,
+    this.scrollController,
+    this.formData,
+  });
+
+  /// The error being displayed
+  final dynamic error;
+
+  /// Padding of the Items in the Form. If no Padding is provided a EdgeInsets.all(8.0) will be used.
+  final EdgeInsetsGeometry padding;
+
+  /// Triggers when the user taps the back button
+  final Function() didTapBackButton;
+
+  /// Optional ScrollController for the Form
+  final ScrollController? scrollController;
+
+  /// The current form data
+  final FormData? formData;
+
+  @override
+  Widget build(BuildContext context) {
+    final localization = ApptiveGridLocalization.of(context)!;
+    final theme = Theme.of(context);
+    return ListView(
+      controller: scrollController,
+      padding: const EdgeInsets.all(32.0),
+      children: [
+        AspectRatio(
+          aspectRatio: 1,
+          child: Lottie.asset(
+            'packages/apptive_grid_form/assets/error.json',
+            repeat: false,
+          ),
+        ),
+        Text(
+          localization.errorTitle,
+          textAlign: TextAlign.center,
+          style: theme.textTheme.headlineMedium,
+        ),
+        Padding(
+          padding: padding,
+          child: Text(
+            error is http.Response
+                ? '${error.statusCode}: ${error.body}'
+                : error.toString(),
+            textAlign: TextAlign.center,
+            style: theme.textTheme.labelSmall
+                ?.copyWith(color: theme.colorScheme.error),
+          ),
+        ),
+        Center(
+          child: TextButton(
+            onPressed: didTapBackButton,
+            child: Text(localization.backToForm),
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Displays a success message after a succesful submission
+class SuccessfulSubmitWidget extends StatelessWidget {
+  /// Creates a new success widget
+  const SuccessfulSubmitWidget({
+    super.key,
+    required this.didTapAdditionalAnswer,
+    this.scrollController,
+    this.formData,
+  });
+
+  /// Triggers when the user taps the button to submit an additional answer
+  final Function() didTapAdditionalAnswer;
+
+  /// Optional ScrollController for the Form
+  final ScrollController? scrollController;
+
+  /// The current form data
+  final FormData? formData;
+
+  @override
+  Widget build(BuildContext context) {
+    final localization = ApptiveGridLocalization.of(context)!;
+    return ListView(
+      controller: scrollController,
+      padding: const EdgeInsets.all(32.0),
+      children: [
+        AspectRatio(
+          aspectRatio: 1,
+          child: Lottie.asset(
+            'packages/apptive_grid_form/assets/success.json',
+            repeat: false,
+          ),
+        ),
+        Text(
+          formData?.properties?.successTitle ?? localization.sendSuccess,
```

Comment:
I think it would be cleaner to pass in the labels for title, message, label as `String (String? for message)` and remove more of the logic from this widget itself meaning this would also not need the form data passed in

Reply:
done